packages/core/src/modules/connections/errors/ConnectionProblemReportError.ts

```diff
@@ -0,0 +1,22 @@
+import type { ConnectionProblemReportReason } from '.'
+import type { ProblemReportErrorOptions } from '../../problem-reports'
+
+import { AriesFrameworkError } from '../../../error/AriesFrameworkError'
+import { ConnectionProblemReportMessage } from '../messages'
+
+interface ConnectionProblemReportErrorOptions extends ProblemReportErrorOptions {
+  problemCode: ConnectionProblemReportReason
+}
+export class ConnectionProblemReportError extends AriesFrameworkError {
```

Comment:
I think we should create a generic `ProblemReportError` where the `ConnectionProblemReportError` can extend from. This way we can check just for `instanceof ProblemReportError` in the dispatcher. As the `ConectionProblemReportMessage` extends the `ProblemReportMessage` it is also possible to use the connection specific problem report message for the `problemReport` property